src/xpcc/driver/radio/dw1000/dw1000_impl.hpp

```diff
@@ -0,0 +1,1294 @@
+		return SUCCESS;
+	}
+	else
+	{
+		return ERROR;
+	}
+}
+
+
+
+template < typename Spi, typename Cs, typename Reset, typename Irq >
+void
+xpcc::Dw1000< Spi, Cs, Reset, Irq >::writetxfctrl(uint16_t txFrameLength, uint16_t txBufferOffset, int ranging)
+{
+	uint32_t reg32 = dw1000local.txFCTRL | txFrameLength | (txBufferOffset << TX_FCTRL_TXBOFFS_SHFT) | (ranging << TX_FCTRL_TR_SHFT);
+	write32bitreg(TX_FCTRL_ID, reg32);
+}
+
+template < typename Spi, typename Cs, typename Reset, typename Irq >
+void
+xpcc::Dw1000< Spi, Cs, Reset, Irq >::write8bitoffsetreg(int regFileID, int regOffset, uint8_t regval)
+{
+	writetodevice(regFileID, regOffset, 1, &regval);
+}
+
+template < typename Spi, typename Cs, typename Reset, typename Irq >
+int
+xpcc::Dw1000< Spi, Cs, Reset, Irq >::starttx(TX_Mode mode)
+{
+	/*int retval = SUCCESS ;
+	uint8_t temp  = 0x00;
+
+	temp |= (uint8_t)SYS_CTRL_TXSTRT ;
+	write8bitoffsetreg(SYS_CTRL_ID, SYS_CTRL_OFFSET, temp);
+
+
+	return retval;*/
+	int retval = SUCCESS ;
+	uint8_t temp  = 0x00;
+	uint16_t checkTxOK = 0 ;
+
+	if(mode & RESPONSE_EXPECTED)
+	{
+		temp = (uint8_t)SYS_CTRL_WAIT4RESP ; // Set wait4response bit
+		write8bitoffsetreg(SYS_CTRL_ID, SYS_CTRL_OFFSET, temp);
+		//dw1000local.wait4resp = 1;
+	}
+
+	if (mode & START_TX_DELAYED)
+	{
+		// Both SYS_CTRL_TXSTRT and SYS_CTRL_TXDLYS to correctly enable TX
+		temp |= (uint8_t)(SYS_CTRL_TXDLYS | SYS_CTRL_TXSTRT) ;
+		write8bitoffsetreg(SYS_CTRL_ID, SYS_CTRL_OFFSET, temp);
+		checkTxOK = read16bitoffsetreg(SYS_STATUS_ID, 3); // Read at offset 3 to get the upper 2 bytes out of 5
+		if ((checkTxOK & SYS_STATUS_TXERR) == 0) // Transmit Delayed Send set over Half a Period away or Power Up error (there is enough time to send but not to power up individual blocks).
+		{
+			retval = SUCCESS ; // All okay
+		}
+		else
+		{
+			// I am taking DSHP set to Indicate that the TXDLYS was set too late for the specified DX_TIME.
+			// Remedial Action - (a) cancel delayed send
+			temp = (uint8_t)SYS_CTRL_TRXOFF; // This assumes the bit is in the lowest byte
+			write8bitoffsetreg(SYS_CTRL_ID, SYS_CTRL_OFFSET, temp);
+			// Note event Delayed TX Time too Late
+			// Could fall through to start a normal send (below) just sending late.....
+			// ... instead return and assume return value of 1 will be used to detect and recover from the issue.
+			//dw1000local.wait4resp = 0;
+			retval = ERROR ; // Failed !
+		}
+	}
+	else
+	{
+		temp |= (uint8_t)SYS_CTRL_TXSTRT ;
+		write8bitoffsetreg(SYS_CTRL_ID, SYS_CTRL_OFFSET, temp);
+	}
+
+	return retval;
+}
+
+template < typename Spi, typename Cs, typename Reset, typename Irq >
+void
+xpcc::Dw1000< Spi, Cs, Reset, Irq >::rxEnable()
+{
+	rxenable(START_RX_IMMEDIATE);
+}
+
+
+template < typename Spi, typename Cs, typename Reset, typename Irq >
+bool
+xpcc::Dw1000< Spi, Cs, Reset, Irq >::rxEnableAt(uint32_t time)
+{
+	setdelayedtrxtime(time);
+	int result;
+	result  = rxenable(1);
```

Comment:
CircleCI fails with
```
src/xpcc/driver/radio/dw1000/dw1000_impl.hpp:853:22: error: no matching function for call to 'xpcc::Dw1000<Spi, Cs, Reset, Irq>::rxenable(int)'
src/xpcc/driver/radio/dw1000/dw1000.hpp:434:2: note: candidate: static int xpcc::Dw1000<Spi, Cs, Reset, Irq>::rxenable(xpcc::dw1000::RX_Mode)
```
This line should probably read as follows
```c++
result = rxenable(RX_Mode::START_RX_DELAYED);
```
([`RX_Mode::START_RX_DELAYED == 1`](https://github.com/Maju-Ketchup/xpcc/blob/feature/DecaWaveDWM1000Driver/src/xpcc/driver/radio/dw1000/params.hpp#L137))

Reply:
You are right. I changed using INTs to a custom enum some time ago. Seems to be a leftover.